packages/elephantdsp-roomreverb/PKGBUILD

```
@@ -0,0 +1,85 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Christopher Arndt <aur -at- chrisarndt -dot- de>
+
+_plugin_name=RoomReverb
+pkgname="elephantdsp-${_plugin_name,,}"
+_plugin_uri="https://www.ElephantDSP.com/plugins/Room_Reverb"
+pkgver=0.6.0
+pkgrel=1
+pkgdesc='A mono and stereo algorithmic room reverb audio effect plugin'
+arch=(x86_64 aarch64)
+url='https://www.elephantdsp.com/products/room-reverb/'
+license=(GPL3)
+depends=(glibc gcc-libs)
+makedepends=(cmake freetype2 'juce>=7' libx11 libxcursor libxinerama libxrandr)
+checkdepends=(lv2lint lv2)
+optdepends=(
+  'clap-host: for CLAP plugin'
+  'lv2-host: for LV2 plugin'
+  'vst3-host: for VST3 plugin'
+)
+groups=(clap-plugins lv2-plugins pro-audio vst-plugins)
+_clap_commit='395ab2bc5abd76f613b7ac3e4292b4e315700749'
+_clap_ext_commit='ff12b486caa173db824691f141b34aad0770d750'
+_clap_helpers_commit='b101259ae06964c76c6806f02ec07847acb7b6dc'
+source=("$pkgname-$pkgver.tar.gz::https://github.com/cvde/RoomReverb/archive/refs/tags/v$pkgver.tar.gz"
+        "clap-juce-extensions-$_clap_ext_commit.zip::https://github.com/free-audio/clap-juce-extensions/archive/$_clap_ext_commit.zip"
+        "clap-$_clap_commit.zip::https://github.com/free-audio/clap/archive/$_clap_commit.zip"
+        "clap-helpers-$_clap_helpers_commit.zip::https://github.com/free-audio/clap-helpers/archive/$_clap_helpers_commit.zip"
+        "roomreverb-devendor_juce.patch")
+sha256sums=('43e2c0d9c47357995543e4731936f6b7167dd441993de1e2d2922a8cd942e2f6'
+            'dbc44e82703781741551499f9f41512aaaa315c588624c3df03eb728ef9e28cf'
+            'dfae419120021a78ba0f322194bbf2ac1eafdee8cbe662bb98e0922f3469e8a3'
+            '967e73fbbd9a6e478c31830da95f45812504c010b74a5295eda1b9a5a8832125'
+            'a7478a0113128e55fdc14e4b861f95bd65c1c64cc20c0d544bce07f90c0a6ede')
+
+
+prepare() {
+  cd $_plugin_name-$pkgver
+
+  # fix for cmake trying to install into ~/.clap, ~/.lv2, and ~/.vst3 during build
+  sed -i -e '/COPY_PLUGIN_AFTER_BUILD/d' CMakeLists.txt
+
+  patch -p1 -N -r - -i "$srcdir"/roomreverb-devendor_juce.patch
+
+  # link external libs
+  cd Libs
+  rm -rf clap-juce-extensions && \
+    ln -s "$srcdir"/clap-juce-extensions-$_clap_ext_commit clap-juce-extensions
+  cd clap-juce-extensions/clap-libs
+  rm -rf clap && \
+    ln -s "$srcdir"/clap-$_clap_commit clap
+  rm -rf clap-helpers && \
+    ln -s "$srcdir"/clap-helpers-$_clap_helpers_commit clap-helpers
+}
+
+build() {
+  cd $_plugin_name-$pkgver
+  cmake \
+    -DCMAKE_BUILD_TYPE=Release \
+    -B build
```

[Arch CMake package guidelines](https://wiki.archlinux.org/title/CMake_package_guidelines) suggest using `-Wno-dev` and a build dir outside the extracted source:
```suggestion
  cmake -S $_plugin_name-$pkgver \
    -DCMAKE_BUILD_TYPE=Release \
    -Wno-dev \
    -B build
```
Then you'll need to change the `cd`s down in `check()` and `package()` as well.

---

Fixed in cfa4111.